src/Middleware/ConvertErrorToJsonResponse.php

```diff
@@ -24,41 +24,60 @@ class ConvertErrorToJsonResponse
      * @return ResponseInterface
      */
     public function __invoke($error, ServerRequestInterface $request, ResponseInterface $response, callable $next): ResponseInterface
+    {
+        return new JsonResponse(self::buildErrorBody($error), self::getStatusCode($error));
+    }
+
+    public static function buildErrorBody($error): array
     {
         switch ($error) {
             case $error instanceof DomainException:
-                return $this->convertDomainException($error);
+                return self::buildDomainExceptionBody($error);
             case $error instanceof Throwable:
-                return $this->convertThrowable($error);
+                return self::buildThrowableBody($error);
             default:
-                return $this->convertGenericError($error);
+                return self::buildGenericErrorBody($error);
         }
     }
 
-    private function convertGenericError($error): JsonResponse
+    public static function getStatusCode($error): int
```

Comment:
Why static? Why public?

Reply:
I chose static because it didn't seem necessary to create a `ConvertErrorToJsonResponse` object just to get the body and status code.

We need it to be public so that the CaptchaService doesn't have to think which statusCode to put for the JsonResponse. The CaptchaService just needs to add `"captcha": true` to the body.

Reply:
This is the intended use `return new JsonResponse(ConvertErrorToJsonResponse::buildErrorBody($domainException), ConvertErrorToJsonResponse::getStatusCode($domainException));`

But if you do not want static function, it can be changed.